src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -219,6 +219,8 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, HttpsConn
             var loggerFactory = listenOptions.KestrelServerOptions?.ApplicationServices.GetRequiredService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
 
             listenOptions.IsTls = true;
```

Comment:
Does IsTls become HttpsOptions != null?

Reply:
The UseHttps overloads that setup a `ServerSelectionCallback` don't set this field.

Reply:
That seems like a bug in those overloads?

Reply:
I mean `HttpsOptions`. It will still be null because its creation is delayed until the connection is created.

Still need IsTls for that situation. In the future I think the callback will also be capture in which case: `public bool IsTls => HttpsOptions != null || HttpOptionsCallback != null;`